### DISCUSSION TITLE
Run PhotoPrism CLI commands inside LXC?

### DISCUSSION THREAD

**Discussion by TwoPuncakes** [tteck/Proxmox#2138] [Help Needed] [answered]
Hello! Is there a way to run PhotoPrism commands within the LXC? From what I understand, these commands can only be ran inside Docker Compose? Is that the case?
- https://docs.photoprism.app/user-guide/users/cli/
- https://docs.photoprism.app/getting-started/docker-compose/#command-line-interface

Here's what happens when I run the help command:
![image](https://github.com/tteck/Proxmox/assets/36810363/0f9df6c9-3029-40c6-8d5c-7930bf12003b)

I'm kind of new to this, so I might be asking a dumb question. Sorry!

**Comment by tteck** [comment]
> PhotoPrism commands

Those are Docker commands

**Comment by rlimaeco** [comment]
The solution for this is to use a symlink like:

sudo ln -sf /opt/photoprism/bin/photoprism /usr/local/bin/photoprism

and you can use as a command, the problem is that everything I tried to run I got this error:

`WARN[2024-01-17T17:42:42+01:00] config: invalid originals path, check configuration and permissions in plus extension [2.339772ms] 
ERRO[2024-01-17T17:42:42+01:00] invalid originals path, check configuration and permissions `

and commands like create a user , or photoprism users ls. but all of them will fail in my case :/

**Reply by tteck** [reply]
What exactly are you trying to do?

**Comment by rlimaeco** [comment]
Just create a user using cli while I don't have a plus subscription

The command tried is

photoprism users add ....

Em qua., 17 de jan. de 2024, 17:52, tteckster ***@***.***>
escreveu:

**Reply by tteck** [reply]
As far as I know, there's no such thing. https://docs.photoprism.app/user-guide/users/

**Reply by TwoPuncakes** [reply]
Huh, never thought this thread would be revived again 😅

Anyway, those web UI options will only appear if we have paid for the PhotoPrism Plus subscription:

> [PhotoPrism® Plus](https://www.photoprism.app/editions#compare) includes a web user interface for account and session management, in addition to the [command-line interface](https://docs.photoprism.app/user-guide/users/cli/) available in all editions.

For free users, the [command-line interface](https://docs.photoprism.app/user-guide/users/cli/) can apparently still create/manage users:

>Managing User Accounts[¶](https://docs.photoprism.app/user-guide/users/cli/#managing-user-accounts)
As an alternative to the [web user interface](https://docs.photoprism.app/user-guide/users/), you can [run the following commands in a terminal](https://docs.photoprism.app/getting-started/docker-compose/#command-line-interface) to perform tasks such as adding, viewing, editing and deleting user accounts:

From what I understand on that page, it seems the commands can only be ran only from Docker.

**Comment by rlimaeco** [comment]
Yes that is my problem, all the cli commands fail when using on this lxc
container :/

Em qui., 18 de jan. de 2024, 09:34, Puncake ***@***.***>
escreveu:

**Reply by TwoPuncakes** [reply]
Pretty unfortunate, but I guess we could use tteck's script to create a Docker LXC and build PhotoPrism from there. Doesn't look too bad, but it'll be a bit tedious.

Or simply just pay and support the PhotoPrism devs :)

**Comment by gunverth** [comment]
If you run

`set -o allexport; source /var/lib/photoprism/.env; set +o allexport`

in the LXC console, you can run all CLI commands including user managing (https://docs.photoprism.app/user-guide/users/cli/#managing-user-accounts) through

`/opt/photoprism/bin/./photoprism`

**Reply by TwoPuncakes** [reply]
I started a fresh PhotoPrism LXC using the script and its default settings and got an error: 
![image](https://github.com/tteck/Proxmox/assets/36810363/2ad982ed-8e47-48ca-99ed-bc210e926346)

**Reply by gunverth** [reply]
OK. Worked for me. Maybe the @tteck script places photoprism in /etc/photoprism. Try change
`/var/lib/photoprism/`
to
`/etc/photoprism/`
in the command

**Reply by TwoPuncakes** [reply]
Still not working unfortunately:
![image](https://github.com/tteck/Proxmox/assets/36810363/d5df1a3a-d96e-4f7f-a655-da718aa7a76d)

I also don't see a `photoprism` folder in `/var/lib/` and `/etc/`
![image](https://github.com/tteck/Proxmox/assets/36810363/780fa37b-6d0d-4bbb-b64b-882373fecdac)

**Reply by TwoPuncakes** [reply]
I looked into [the script](https://github.com/tteck/Proxmox/blob/main/install/photoprism-install.sh) and saw where the .env file is located:
```
[Service]
Type=forking
User=root
WorkingDirectory=/opt/photoprism
EnvironmentFile=/opt/photoprism/config/.env
ExecStart=/opt/photoprism/bin/photoprism up -d
ExecStop=/opt/photoprism/bin/photoprism dow
```
So then I ran `set -o allexport; source /opt/photoprism/config/.env; set +o allexport` instead and seemed to have went through:
![image](https://github.com/tteck/Proxmox/assets/36810363/3401d694-1c92-4cef-a399-70259337da89)

Then I went to the bin folders to run the commaand, but ran into more issues:
![image](https://github.com/tteck/Proxmox/assets/36810363/59b0682a-ab93-41a3-8ebb-8d86e7158978)

Honestly, not 100% sure what I'm doing at this point lol

**Reply by gunverth** [reply]
Try
`./photoprism <command>`
Also; the PhotoPrism package for Debian puts the "data directory" in /var/lib/photoprism/ since /opt/photoprism is used for binaries which will be replaced when upgraded.

**Reply by TwoPuncakes** [reply, marked as answer]
I see, now it works! 

![image](https://github.com/tteck/Proxmox/assets/36810363/f9c7e451-dfea-42e5-91bd-c07a1f943a63)

So the commands to run (for me at least) are:
```
set -o allexport; source /opt/photoprism/config/.env; set +o allexport
cd /opt/photoprism/bin/
./photoprism <command>
```

Thanks so much for looking into this!

**Reply by rlimaeco** [reply]
Thanks a lot , this worked pretty well ! Best Regards !
